docs/src/pages/guides/migrating-to-react-query-3.md

```diff
@@ -116,6 +116,46 @@ Together, these provide the same experience as before, but with added control to
 
 **Notice that it's now a function instead of a property**
 
+### `QueryStatus` has been changed from an [enum](https://www.typescriptlang.org/docs/handbook/enums.html#string-enums) to a [union type](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types)
+
+So, if you were checking the status property of a query or mutation against a QueryStatus enum property you will have to check it now against the string literal the enum previously held for each property.
+
+Therefore you have to change the enum properties to their equivalent string literal, like this:
+- `QueryStatus.Idle` -> `'idle'`
+- `QueryStatus.Loading` -> `'loading'`
+- `QueryStatus.Error` -> `'error'`
+- `QueryStatus.Success` -> `'success'`
+
+This is an example of how you would have done it in the previous version:
+
+```js
+import { QueryStatus } from 'react-query';
+
+const { data, status } = useQuery(['post', id], (_key, id) => fetchPost(id))
+
+if (status === QueryStatus.Loading) {
+  ...
+}
+
+if (status === QueryStatus.Error) {
+  ...
+}
+```
+
+And this is how it is now in version 3:
+
+```js
+const { data, status } = useQuery(['post', id], () => fetchPost(id))
+
+if (status === 'loading') {
+  ...
+}
+
+if (status === 'error') {
+  ...
+}
+```
```

Comment:
maybe we can show a `diff`:

```diff
- import { useQuery, QueryStatus } from 'react-query'
+ import { useQuery } from 'react-query'

const { data, status } = useQuery(['post', id], (_key, id) => fetchPost(id))

- if (status === QueryStatus.Loading) {
+ if (status === 'loading') {
  ...
}

- if (status === QueryStatus.Error) {
+ if (status === 'error') {
  ...
}
```

Reply:
Sure, we will have to show the diff on the query function too right? now it's given a context as a parameter not the query keys

Reply:
ah yes, I missed that!

Reply:
I'm a bit hesitant about the diff, I don't know if it seems a bit cluttered the example or if it's just me, it looks fine anyways
![image](https://user-images.githubusercontent.com/3933427/113449023-3fb88b80-93fd-11eb-9c09-04a519a311e3.png)

Reply:
right, maybe we could get rid of the difference for the actual useQuery call? the `(_key, id)` change is not actually part of the union type change, so we can just use the inline function for both version? Then it looks good I'd say :) I updated the diff here: https://github.com/tannerlinsley/react-query/pull/2073#discussion_r606296742

Reply:
Done :+1: